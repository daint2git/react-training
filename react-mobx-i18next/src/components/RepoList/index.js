import React from 'react'
import { observer } from 'mobx-react'
import { withTranslation } from 'react-i18next'

const RepoList = ({ repos, repoCount }) => (
  <>
    <ul>
      {repos.map(repo => (
        <li key={repo.id}>{repo.htmlUrl}</li>
      ))}
    </ul>
    <p>Have {repoCount} repos.</p>
  </>
)

export default withTranslation()(observer(RepoList))
