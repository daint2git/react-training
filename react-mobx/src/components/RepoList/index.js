import React from 'react'
import { inject, observer } from 'mobx-react'

@inject('RepoStore')
@observer
class RepoList extends React.Component {
  componentDidMount() {
    const {
      RepoStore: { fetchRepos },
    } = this.props

    fetchRepos()
  }

  render() {
    const {
      RepoStore: { repos, repoCount },
    } = this.props

    return (
      <div>
        <h2>Repos</h2>
        <ul>
          {repos.map(repo => (
            <li key={repo.id}>{repo.htmlUrl}</li>
          ))}
        </ul>
        <p>Have {repoCount} repos.</p>
      </div>
    )
  }
}

export default RepoList
