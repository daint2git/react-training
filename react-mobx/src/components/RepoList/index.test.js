import RepoList from './'

const renderComponentWithStore = store => mount(<RepoList RepoStore={store} />)

describe('RepoList', () => {
  let wrapper

  afterEach(() => {
    wrapper.unmount()
  })

  test('should render correctly default data', () => {
    const mockStore = {
      // observable
      repos: [],
      repoCount: 0,
      // actions
      fetchRepos: jest.fn(),
    }
    wrapper = renderComponentWithStore(mockStore)
    expect(wrapper).toMatchSnapshot()
  })

  test('should render correctly mock data', () => {
    const repos = [
      {
        id: 1,
        htmlUrl: 'abc1',
      },
      {
        id: 2,
        htmlUrl: 'abc2',
      },
    ]

    const mockStore = {
      // observable
      repos,
      repoCount: repos.length,
      // actions
      fetchRepos: jest.fn(),
    }

    wrapper = renderComponentWithStore(mockStore)
    expect(wrapper).toMatchSnapshot()
  })

  test('should call fetchRepos callback', () => {
    const mockStore = {
      // observable
      repos: [],
      repoCount: 0,
      // actions
      fetchRepos: jest.fn(),
    }
    wrapper = renderComponentWithStore(mockStore)
    expect(mockStore.fetchRepos).toHaveBeenCalled()
    expect(mockStore.fetchRepos).toHaveBeenCalledTimes(1)
  })
})
