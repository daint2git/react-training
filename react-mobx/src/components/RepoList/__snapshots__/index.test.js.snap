// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoList should render correctly default data 1`] = `
<div>
  <h2>
    Repos
  </h2>
  <ul />
  <p>
    Have 
    0
     repos.
  </p>
</div>
`;

exports[`RepoList should render correctly mock data 1`] = `
<div>
  <h2>
    Repos
  </h2>
  <ul>
    <li
      key="1"
    >
      abc1
    </li>
    <li
      key="2"
    >
      abc2
    </li>
  </ul>
  <p>
    Have 
    2
     repos.
  </p>
</div>
`;
